vkcs/dc/resource_router.go

```diff
@@ -0,0 +1,280 @@
+package dc
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/clients"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/dc/v2/routers"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/networking"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/util"
+)
+
+// Ensure the implementation satisfies the desired interfaces.
+var _ resource.Resource = &RouterResource{}
+
+func NewRouterResource() resource.Resource {
+	return &RouterResource{}
+}
+
+type RouterResource struct {
+	config clients.Config
+}
+
+func (r *RouterResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "vkcs_dc_router"
+}
+
+type RouterResourceModel struct {
+	ID               types.String `tfsdk:"id"`
+	AvailabilityZone types.String `tfsdk:"availability_zone"`
+	Flavor           types.String `tfsdk:"flavor"`
+	Name             types.String `tfsdk:"name"`
+	Description      types.String `tfsdk:"description"`
+	CreatedAt        types.String `tfsdk:"created_at"`
+	UpdatedAt        types.String `tfsdk:"updated_at"`
+	Region           types.String `tfsdk:"region"`
+}
+
+func (r *RouterResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the resource",
+			},
+
+			"availability_zone": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The availability zone in which to create the router. Changing this creates a new router",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"flavor": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "Flavor of the router. Possible values can be obtained with vkcs_dc_api_options data source. Not to be confused with compute service flavors. Changing this creates a new router",
```

Comment:
Lets mark "not to be confused..." as note.

Reply:
Marked